src/main/java/de/tum/in/www1/artemis/service/GroupNotificationService.java

```diff
@@ -7,220 +7,145 @@
-        String notificationText = "Exercise \"" + exercise.getTitle() + "\" got a new question.";
-        User user = userService.getUser();
-        GroupNotification groupNotification = new GroupNotification(exercise.getCourse(), title, notificationText, user, type);
-        groupNotification.setTarget(groupNotification.getExerciseQuestionTarget(exercise));
-        return groupNotification;
-    }
-
-    private GroupNotification createLectureQuestionCreatedGroupNotification(StudentQuestion studentQuestion, GroupNotificationType type) {
-        Lecture lecture = studentQuestion.getLecture();
-        String title = "New Question";
-        String notificationText = "Lecture \"" + lecture.getTitle() + "\" got a new question.";
-        User user = userService.getUser();
-        GroupNotification groupNotification = new GroupNotification(lecture.getCourse(), title, notificationText, user, type);
-        groupNotification.setTarget(groupNotification.getLectureQuestionTarget(lecture));
-        return groupNotification;
-    }
-
-    private GroupNotification createExerciseAnswerCreatedGroupNotification(StudentQuestionAnswer studentQuestionAnswer, GroupNotificationType type) {
-        Exercise exercise = studentQuestionAnswer.getQuestion().getExercise();
-        String title = "New Answer";
-        String notificationText = "Exercise \"" + exercise.getTitle() + "\" got a new answer.";
-        User user = userService.getUser();
-        GroupNotification groupNotification = new GroupNotification(exercise.getCourse(), title, notificationText, user, type);
-        groupNotification.setTarget(groupNotification.getExerciseAnswerTarget(exercise));
-        return groupNotification;
-    }
-
-    private GroupNotification createLectureAnswerCreatedGroupNotification(StudentQuestionAnswer studentQuestionAnswer, GroupNotificationType type) {
-        Lecture lecture = studentQuestionAnswer.getQuestion().getLecture();
-        String title = "New Answer";
-        String notificationText = "Lecture \"" + lecture.getTitle() + "\" got a new answer.";
-        User user = userService.getUser();
-        GroupNotification groupNotification = new GroupNotification(lecture.getCourse(), title, notificationText, user, type);
-        groupNotification.setTarget(groupNotification.getLectureAnswerTarget(lecture));
-        return groupNotification;
+    /**
+     * Notify student groups about an attachment change.
+     *
+     * @param attachment that has been changed
+     * @param notificationText that should be displayed  TODO: can be removed in the future as notification's text attribute is not used in the client
+     */
+    public void notifyStudentGroupAboutAttachmentChange(Attachment attachment, String notificationText) {
+        // Do not send a notification before the release date of the attachment.
+        if (attachment.getReleaseDate() != null && !attachment.getReleaseDate().isBefore(ZonedDateTime.now())) {
```

Comment:
I think this would be easier to parse mentally:

```suggestion
        if (attachment.getReleaseDate() != null && attachment.getReleaseDate().isAfter(ZonedDateTime.now())) {
```

Reply:
Definitely!